src/main/java/org/gridsuite/study/server/dto/impacts/AbstractBaseImpact.java

```diff
@@ -26,22 +26,17 @@
     visible = true
 )
 @JsonSubTypes({
-    @JsonSubTypes.Type(value = SimpleElementImpact.class, names = { "CREATION", "MODIFICATION", "DELETION" }),
+    @JsonSubTypes.Type(value = SimpleElementImpact.class, name = "SIMPLE"),
     @JsonSubTypes.Type(value = CollectionElementImpact.class, name = "COLLECTION")
 })
-@AllArgsConstructor(access = AccessLevel.PRIVATE)
-@Setter
-@Getter
 @SuperBuilder
+@Data
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
 @NoArgsConstructor
-@EqualsAndHashCode(callSuper = false)
-@ToString
 public abstract class AbstractBaseImpact {
 
     public enum ImpactType {
-        CREATION,
-        MODIFICATION,
-        DELETION,
+        SIMPLE,
         COLLECTION
     }
```

Comment:
This enum and the _impactType_ field are redundant because there is already 2 different classes for them, so type introspection should do the job.